src/config.js

```diff
@@ -257,13 +257,20 @@ export default class Config {
    */
 
   getOfflineMirrorPath(tarUrl: ?string): ?string {
-    const registry = this.registries.npm;
-    if (registry == null) {
-      return null;
+    //
+    let mirrorPath;
+
+    for (const key of Object.keys(this.registries)) {
+      const registry = this.registries[key];
+      const registryMirrorPath = registry.config['yarn-offline-mirror'];
+
+      if (registryMirrorPath == null) {
+        continue;
```

Comment:
This answers the last question - I also do think it makes sense.
Though, it introduces the `registry == null` check again, as we can't be sure that both registries do exist, am I right?